drivers/sensor/amg88xx/amg88xx.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2017 Phytec Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <device.h>
+#include <i2c.h>
+#include <gpio.h>
+#include <misc/byteorder.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <sensor.h>
+#include <misc/__assert.h>
+
+#include "amg88xx.h"
+
+int amg88xx_reg_read(struct amg88xx_data *drv_data, u8_t reg, u8_t *val)
+{
+	return i2c_reg_read_byte(drv_data->i2c, CONFIG_AMG88XX_I2C_ADDR,
+				 reg, val);
+}
+
+int amg88xx_reg_write(struct amg88xx_data *drv_data, u8_t reg, u8_t val)
+{
+	return i2c_reg_write_byte(drv_data->i2c, CONFIG_AMG88XX_I2C_ADDR,
+				  reg, val);
+}
+
+int amg88xx_reg_update(struct amg88xx_data *drv_data, u8_t reg,
+		     u8_t mask, u8_t val)
+{
+	return i2c_reg_update_byte(drv_data->i2c, CONFIG_AMG88XX_I2C_ADDR,
+				   reg, mask, val);
+}
+
+static int amg88xx_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct amg88xx_data *drv_data = dev->driver_data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_TEMP);
+
+	if (i2c_burst_read(drv_data->i2c, CONFIG_AMG88XX_I2C_ADDR,
+			   AMG88XX_OUTPUT_BASE,
+			   (u8_t *)drv_data->sample, 128)) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int amg88xx_channel_get(struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct amg88xx_data *drv_data = dev->driver_data;
+	size_t len = ARRAY_SIZE(drv_data->sample);
+
+	if (chan != SENSOR_CHAN_TEMP) {
+		return -ENOTSUP;
+	}
+
+	for (size_t idx = 0; idx < len; idx++) {
+		/* fix negative values */
+		if (drv_data->sample[idx] & (1 << 11)) {
+			drv_data->sample[idx] |= 0xF000;
+		}
+		val[idx].val1 = (((s32_t)drv_data->sample[idx]) *
+				  AMG88XX_TREG_LSB_SCALING) / 100;
+		val[idx].val2 = (((s32_t)drv_data->sample[idx]) *
+				  AMG88XX_TREG_LSB_SCALING) % 100;
+	}
+
+	return 0;
+}
+
+int amg88xx_init_device(struct device *dev)
+{
+	struct amg88xx_data *drv_data = dev->driver_data;
+	u8_t tmp;
+
+	if (amg88xx_reg_read(drv_data, AMG88XX_PCLT, &tmp)) {
+		return -EIO;
+	}
+
+	SYS_LOG_DBG("Power mode 0x%02x", tmp);
+	if (tmp != AMG88XX_PCLT_NORMAL_MODE) {
+		if (amg88xx_reg_write(drv_data, AMG88XX_PCLT,
+				       AMG88XX_PCLT_NORMAL_MODE)) {
+			return -EIO;
+		}
+		k_busy_wait(AMG88XX_WAIT_MODE_CHANGE_US);
+	}
+
+	if (amg88xx_reg_write(drv_data, AMG88XX_RST,
+			       AMG88XX_RST_INITIAL_RST)) {
+		k_busy_wait(AMG88XX_WAIT_INITIAL_RESET_US);
```

Comment:
Why is this here? Shouldn't it be after the if statement (i. e., if the write is successful)?